requirements.txt

```diff
@@ -3,10 +3,11 @@ RandomWords
 apscheduler==3.5.1
 argcomplete
 bcrypt
-cryptography==2.2.2
+cryptography==2.3.0
 eventlet==0.23.0
 flex==6.13.1
 git+https://github.com/Kami/logshipper.git@stackstorm_patched#egg=logshipper
+git+https://github.com/StackStorm/mongoengine.git@stackstorm_patched#egg=mongoengine
```

Comment:
I'm ok with using a fork only if our patch is going to merge upstream in the near term. But if this is indefinitely, this is harder to agree because we will have to maintain this fork moving forward.

Reply:
The problem is that I'm not really aware of a good upstream fix (besides reverting that change).

I wish there was a better way, but it doesn't look like there is one and the performance regression is sadly too large for us to simply ignore it.

"Good" news is that mongoengine has relatively slow release cadence so keeping our fork up to date shouldn't be too bad (keep in mind that we haven't upgraded mongoengine in our repo for 1 year because of that regression).